### PR TITLE
[Refactor RetinaNet] Step 7, unify nms decoder outputs for both models.

### PR DESCRIPTION
**Pull request by tanzhenyu**
Will need to fix nms unit test

### CODE REVIEW

File: keras_cv/layers/object_detection/non_max_suppression.py [hunk 1]
```diff
@@ -129,53 +134,18 @@ def call(self, predictions, images=None):
...
-
-    def _encode_to_ragged(self, boxes, valid_detections):
-        # using cumulative sum to calculate row_limits for ragged tensor
-        row_limits = tf.cumsum(valid_detections)
-        # creating the output RaggedTensor by splitting boxes at row_limits
-        result = tf.RaggedTensor.from_row_limits(values=boxes, row_limits=row_limits)
-        return result
+        return {
+            "boxes": box_pred,
+            "scores": scores_pred,
```
Comment (LukeWood) [inline keras_cv/layers/object_detection/non_max_suppression.py:145]: scores => confidence

File: keras_cv/layers/object_detection/non_max_suppression.py [hunk 2]
```diff
@@ -129,53 +134,18 @@ def call(self, predictions, images=None):
...
-        # stacking to create a tensor
-        return boxes.stack()
-
-    def _encode_to_ragged(self, boxes, valid_detections):
-        # using cumulative sum to calculate row_limits for ragged tensor
-        row_limits = tf.cumsum(valid_detections)
-        # creating the output RaggedTensor by splitting boxes at row_limits
-        result = tf.RaggedTensor.from_row_limits(values=boxes, row_limits=row_limits)
-        return result
+        return {
```
Comment (LukeWood) [inline keras_cv/layers/object_detection/non_max_suppression.py:143]: convert_format() missing

**Review by LukeWood — APPROVED** [review]
LGTM with an update from confidence-> scores and an update. to unit tests

File: keras_cv/layers/object_detection/non_max_suppression.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection/non_max_suppression.py]: Good point. Done

**Comment by LukeWood** [comment]
you'll also need to fix the prediction decoder test

**Pull request merged**